source/GStreamerMSEMediaPlayerClient.cpp

```diff
@@ -474,7 +480,11 @@ HaveDataMessage::HaveDataMessage(firebolt::rialto::MediaSourceStatus status, uns
 
 void HaveDataMessage::handle()
 {
-    mPlayer->mClientBackend->haveData(mStatus, mNeedDataRequestId);
+    // Make sure we have a valid media pipeline before sending the data
+    if(mPlayer && mPlayer->mClientBackend)
```

Comment:
It's not multithread safe, because mClientBackend will be reset in different thread than HaveDataMessage::handle() is called.
You might need to stop MessageQueues before calling destroyClientBackend (please check GStreamerMSEMediaPlayerClient's destructor)

Reply:
Reverted